curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/64ae550d9fdb40025e5857ae.md

```diff
@@ -0,0 +1,367 @@
+---
+id: 64ae550d9fdb40025e5857ae
+title: Step 47
+challengeType: 0
+dashedName: step-47
+---
+
+# --description--
+
+You need to run the newly created `straightDetector` function to see the results on the display. To do this, create a function named `findRollResult` and set the argument to `arr`. Then, call the function `straightDetector` with the argument of `arr`.
+
+# --hints--
+
+You should create a function named `findRollResult` with an argument of `arr`.
+
+```js
+assert.match(code, /function\s*findRollResult\s*\(\s*arr\s*\)\s*{/);
+```
+
+You should call the function `straightDetector` with the argument of `arr`.
+
+```js
+assert.match(code, /function\s*findRollResult\s*\(\s*arr\s*\)\s*{\s*straightDetector\s*\(\s*arr\s*\)\s*}/);
```

Comment:
Updated Regex for arrow function

```suggestion
assert.match(code, /findRollResult\s*=\s*\(arr\)\s*=>\s*{\s*straightDetector\(arr\);?/);
```

Reply:
```assert.match(code, /(function\s*findRollResult\s*\(\s*arr\s*\)\s*{\s*straightDetector\s*\(\s*arr\s*\)\s*}|const\s*findRollResult\s*=\s*\(arr\)\s*=>\s*{\s*straightDetector\s*\(\s*arr\s*\)\s*})/);```

this will meet both cases